### PR TITLE
cherry-picking from #114 Pvs analyze part 2

### PR DESCRIPTION
**Pull request by p-alik**
please, review.

### CODE REVIEW

**Comment by esabol** [comment]
Use curly brace initialization style in libgearman-server/struct/gearmand.h lines 53-56?

File: libgearman-server/gearmand_thread.cc [hunk 1]
```diff
@@ -160,19 +160,24 @@ namespace {
...
+  next{nullptr},
+  prev{nullptr},
+  base{nullptr},
+  dcon_list{nullptr},
+  dcon_add_list{nullptr},
+  free_dcon_list{nullptr},
+  server_thread{},
+  id{},
+  lock{}
+  //TODO uniform initialisation doesn't help with wakeup_event
```
Comment (esabol) [inline libgearman-server/gearmand_thread.cc]: What's this TODO?

File: libgearman-server/struct/gearmand.h [hunk 2]
```diff
@@ -75,7 +74,7 @@ struct gearmand_st
 
     void keepalive_idle(int keepalive_idle_)
     {
-      _keepalive= true;
+      keepalive(1);
```
Comment (esabol) [inline libgearman-server/struct/gearmand.h]: Is `_keepalive= true` bad coding style according to PVS? It should be more efficient than `keepalive(1)` unless the compiler can optimize it.

**Comment by esabol** [comment]
Should libgearman-server/struct/thread.h line 64 have curly braces after `lock`?

File: libgearman-server/gearmand_thread.cc (hunk 1 above)
Comment (p-alik) [reply libgearman-server/gearmand_thread.cc]: Unfortunately I can't remember.

File: libgearman-server/struct/gearmand.h (hunk 2 above)
Comment (p-alik) [reply libgearman-server/struct/gearmand.h]: the commit with the changes is removed

File: libgearman-server/gearmand_thread.cc (hunk 1 above)
Comment (esabol) [reply libgearman-server/gearmand_thread.cc]: Probably should remove it then.

**Comment by p-alik** [comment]
> Should libgearman-server/struct/thread.h line 64 have curly braces after `lock`?

I guess the compiler disliked `pthread_mutex_t lock{}`. But `g++ (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609` accepts braces.

**Comment by p-alik** [comment]
> Is there something we can turn on after these changes to make sure we don't backslide?

Improving of test coverage could help, I think.

**Pull request merged**